okpt/test/runner.py

```diff
@@ -0,0 +1,146 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Provides a test runner class."""
+import dataclasses
+import platform
+import sys
+from datetime import datetime
+from typing import Any, Dict, List
+
+import psutil
+
+from okpt.io.config.parsers import base, tool
+from okpt.test.tests import nmslib, opensearch
+
+
+def _get_test(test_id: int):
```

Comment:
As mentioned above, change id from int to unqiue description.

Also, given that this function will change everytime a new test is added, could you place this in a different file? Maybe TestFactory

Reply:
Done